odoo/addons/base/models/res_users.py

```diff
@@ -578,18 +578,20 @@ def authenticate(cls, db, login, password, user_agent_env):
                relevant environment attributes
         """
         uid = cls._login(db, login, password)
-        if uid == ADMINUSER_ID:
-            # Successfully logged in as admin!
-            # Attempt to guess the web base url...
-            if user_agent_env and user_agent_env.get('base_location'):
-                try:
-                    with cls.pool.cursor() as cr:
+        if user_agent_env and user_agent_env.get('base_location'):
+            with cls.pool.cursor() as cr:
+                env = api.Environment(cr, uid, {})
+                admin_user = env.ref('base.user_admin', False)
```

Comment:
Thanks for the PR :-) I think the introduction of a non-root admin is an opportunity to use group-based permission checks in more places. Instead of looking for `base.user_admin`, shouldn't we verify that the user is a member of `base.group_system`? This way users are free to reassign/reorganize admin users as they wish.
The group membership test looks good enough for the base URL, and will automatically work for UID 1 too, in addition to any other admin. What do you think?

Reply:
It woiuld work well of course, but what is the use case exactly for continued updating of the base url? Do you really want to increase the number of updates? I believe the case of UID 1 should not be considered here as this user seems to be set up not to log on to the system.

Reply:
The number of updates would only change marginally, and should not matter much, as long as the value is stable. Issues with concurrent updates are unlikely unless admin users are used by automated scripts - an issue that could have happened with UID 1 before as well. I believe this is solved with the freeze mechanism, similarly to situations with unstable base URLs.
Basically it seems a good idea to avoid relying on a fixed admin here if users are allowed to change it.

Reply:
My qualm was more in increasing the risk that someone tunnels in from another location and messes up the value rather than database concurrency issues, but you are right that that is what the freeze parameter is for. Refactored accordingly!